### PR TITLE
[92X] L1 no Hack Conditions

### PR DESCRIPTION
**Pull request by kkotov**
Removing Hack Conditions from L1Trigger package,
and start using configurations from DB, via GTs.

Ajdustements to AddOn workflows with 2015 data have been made.

Backport of #20577 (made from Vladimir's #20644)

### CODE REVIEW

**Comment by kkotov** [comment]
please test

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -17,10 +17,23 @@
 #                     pset.minGoodStripCharge = cms.PSet(refToPSet_ = cms.string('HLTSiStripClusterChargeCutNone'))
 #     return process
 
+# Migrate uGT non-CondDB parameters to new cff: remove StableParameters dependence in favour of GlobalParameters
+def customiseFor19989(process):
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: Wrong PR number - this should be the 92X PR number 20689

**Comment by arunhep** [comment]
please test

**Comment by kkotov** [comment]
@arunhep, this PR may require backports of PR #20397, #20494, #20518 to have all of the required tags in the 92X GTs. Or were those already backported maybe?

**Comment by Martin-Grunewald** [comment]
@kkotov 
This starts to mushroom - is the "no hack" backport to 92X really needed for data taking, or just "nice to have"?

**Comment by Martin-Grunewald** [comment]
Also, this PR will not get integrated before #20577 is integrated in 94X, but that one no longer merges and seems unattended.

**Comment by kkotov** [comment]
@Martin-Grunewald , the idea was to let DQM run with the O2O conditions. And DQM still uses 92X. This was the reason for the original #20644 that is also seems to be no longer attended.

**Comment by Martin-Grunewald** [comment]
Yeah well but is it really needed (required updates which can not be done with "hack")?

**Comment by kkotov** [comment]
@Martin-Grunewald , there was another thread where DQM people didn't accept our original update (or rather a patch) because it was not part of the CMSSW (and I can understand why they don't want to support such solutions). So we choose to go all the way including this backport. If only I knew how much of the unrelated stuff will be pulled up in the process ... But now I have an impression that we've finally are pretty close to the end, although we indeed loose motivation and people on the way :0)

**Comment by arunhep** [comment]
@kkotov @rekovic 
Can you please cherry-pick this commit : https://github.com/arunhep/cmssw/commit/5e263f356051f891d469fe20c1f8d10fade4d322

and restart the tests.

you are going to have another commit from me later to update other 2017/2018 GTs. But that is just an update of tag and not an addition of tag. The commit proposed above should address the crashes. If not, we will see more.

**Comment by arunhep** [comment]
please test

File: Configuration/AlCa/python/autoCond.py [hunk 2]
```diff
@@ -10,17 +10,19 @@
...
     'run1_mc_pa'        :   '92X_mcRun1_pA_v2',
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Run2
-    'run2_design'       :   '92X_mcRun2_design_v2',
+    'run2_design'       :   '92X_mcRun2_design_v3',
     # GlobalTag for MC production with pessimistic alignment and calibrations for Run2
-    'run2_mc_50ns'      :   '92X_mcRun2_startup_v2',
+    'run2_mc_50ns'      :   '92X_mcRun2_startup_v3',
     #GlobalTag for MC production with optimistic alignment and calibrations for Run2
-    'run2_mc'           :   '92X_mcRun2_asymptotic_v2',
+    'run2_mc'           :   '92X_mcRun2_asymptotic_v3',
```
Comment (arunhep) [inline Configuration/AlCa/python/autoCond.py]: The reason of failures now is this v3. This should be v4.

Comment (kkotov) [reply Configuration/AlCa/python/autoCond.py]: done

**Comment by Martin-Grunewald** [comment]
please test

**Comment by arunhep** [comment]
please test

**Comment by arunhep** [comment]
@kkotov please include this commit https://github.com/arunhep/cmssw/commit/d976e8cf3c0bfd28b739262c93c85f3f16508d07
this will complete the update of L1T tags for 2017/2018 as well.

**Comment by kkotov** [comment]
@arunhep , done

**Comment by arunhep** [comment]
@kkotov also this one https://github.com/arunhep/cmssw/commit/279d629ee605f17df95b18c5c94fff7489e6a137
hopefully the last one. this should address the recent crashes in Phase-II workflows

**Comment by arunhep** [comment]
@cmsbuild please test

**Comment by Martin-Grunewald** [comment]
Remaining error is in the fastsim1 addOnTest:
```
----- Begin Fatal Exception 04-Oct-2017 00:35:06 CEST-----------------------
An exception of category 'OutOfRange' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 3
   [1] Running path 'L1simulation_step'
   [2] Calling method for module L1TPhysicalEtAdder/'simCaloStage1FinalDigis'
Exception Message:
Index out of range in L1CaloEtScale::et(rank)
----- End Fatal Exception -------------------------------------------------
```

**Comment by Martin-Grunewald** [comment]
It seems the fatssim1 test should use ```auto:run2_mc_l1stage1``` in ```Utilities/ReleaseScripts/scripts/addOnTests.py```

**Comment by kkotov** [comment]
thanks for the hint, @Martin-Grunewald

**Comment by arunhep** [comment]
please test

**Comment by arunhep** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by rekovic** [comment]
+1

**Comment by Martin-Grunewald** [comment]
@rekovic @kkotov 
There is now a merge conflict in this PR which must be fixed!

**Comment by Dr15Jones** [comment]
+1
for Utilities/ReleaseScripts

**Comment by arunhep** [comment]
@rekovic @Martin-Grunewald 
Can this conflict be resolved asap so that this PR can be merged?
Then only we can provide updated GTs including V4 L1T menu.
Otherwise, we will have multiple copies of autoCond open and it will be difficult to manage for us if we find some problems in updated conditions.

**Comment by Martin-Grunewald** [comment]
This now becomes a problem as merging this PR would require a further change in HLT menu for online running (can't have customisations there) for which we have no time. Are the GTs useable without this PR, or do they have to go together?

**Comment by arunhep** [comment]
@Martin-Grunewald Sorry but I don't understand the comment. 
According to me, this PR allows the L1T emulator to read conditions from DB via GTs instead of static files. @rekovic @kkotov can clarify more? Apart from resolving the conflict, is there anything else needed for merging of this PR ?

**Comment by Martin-Grunewald** [comment]
@arunhep 
The problem is this required customisation for the HLT menu: https://github.com/cms-sw/cmssw/pull/20689/files#diff-5b29e53ee0c1d396908444c2711d39f6
which is fine offline (to make it work) but not online as there the HLT menu is directly extracted from ConfDB. Thus we'd have to put this customisation directly into the menu, which we'll of course do eventually but preferrably not right now. 

Of course, if online we stay with <= 9_2_13 and do not update to the next 92X release containing this PR, then it would also be fine as it gives us time (HLT v4 runs with current 92 releases, and the online GT gets the v4 L1T menu via O2O). But for offline use with the next 92 release we need the GTs.

Given the long-standing saga of "no-hack", which is not even integrated in 94X (PR there is #20871), I am wary about any timelines to converge on "no hack" in 92X. It is forced now due to GTs, so fine, let's push and hope!

**Comment by rekovic** [comment]
@kkotov please resolve the conflicts.

**Comment by rekovic** [comment]
@arunhep, @Martin-Grunewald 
There is nothing else apart from resolving conflicts. 

@kkotov 
 I would suggest to disentangle changes to GTs in autoCond.py  from this PR to speed up the integration
- revert  changes to Configuration/AlCa/python/autoCond.py
- test with Aruns future AlCA PR which will include the L1TMenu v4, together with the latest BMTF, EMTF, CaloParams payloads.

**Comment by kkotov** [comment]
I am not sure I understand the consequences of a seemingly harmless commit resolving the conflict. Looking at diff of this commit I see ~200 files changed.

**Comment by arunhep** [comment]
@rekovic future PR will only contain L1 menu update on the top of current autoCond here in this PR.
Other latest BMTF, EMTF, CaloParams payloads will be added later.

@Martin-Grunewald Can V4 L1T menu be updated independently of HLT update (means your PR) or it should go with HLT menu update?

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
Thanks @arunhep.
Then, let's try to integrate this PR  now that the conflicts are resolved.

**Comment by rekovic** [comment]
@arunhep 
Or do you prefer us to disentangle the GT parts of this PR?

**Comment by arunhep** [comment]
>Thanks @arunhep.
Then, let's try to integrate this PR now that the conflicts are resolved.

this would be the cleanest

**Comment by rekovic** [comment]
ОК.

**Comment by kkotov** [comment]
@rekovic , trigger the tests again, please. Thx.

**Comment by rekovic** [comment]
please test

**Comment by Martin-Grunewald** [comment]
@arunhep 
Once the GTs contain the v4 L1T, we must switch to the v4 HLT (ie, @relval2017 to point to the V4.0 HLT menu. That menu is already in my PRs but not used yet for relvals).

**Comment by rekovic** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by kkotov** [comment]
@rekovic , following up our IRL discussion, I confirm that wf 145.0 runs fine with this PR.

On the other hand this PR differs from the merged #20871 94X counterpart in
1) **L1Trigger/Configuration/python/SimL1Emulator_cff.py** (added the new EMTF emulator conditions) 
2) a couple of DQM related files where the the GlobalParameters_cff added in place of hackConditions and the new EMTF emulator conditions file is added:
**DQM/Integration/python/clients/l1tstage2emulator_dqm_sourceclient-live_cfg.py**
**DQMOffline/L1Trigger/python/L1TriggerDqmOffline_cff.py**

If I update the the three files above so as DQM can run smooth, you'll have to retrigger the tests.

**Comment by kkotov** [comment]
This PR is also mergeable with #20768 (the new EMTF emulator) with no conflicts. Updating the files that I mentioned earlier makes sense only if the EMTF's PR is merged beforehand, of course.

**Comment by thomreis** [comment]
Hi @kkotov I will let the DQM people know that they have to maintain the order of PRs when they merge them for P5.

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by Martin-Grunewald** [comment]
-1
at this stage there is no point anymore for this to go in 92X

**Comment by rekovic** [comment]
please test

**Comment by arunhep** [comment]
@kkotov its better to close it now. i think this update is out of scope.

**Pull request closed without merging**